java-manta-client-unshaded/src/main/java/com/joyent/manta/http/MantaHttpHeaders.java

```diff
@@ -1065,36 +1065,7 @@ public MantaHttpHeaders setByteRange(final Long start, final Long end) {
      * @return two value array containing the start and the end of a byte range as Long
      */
     public Long[] getByteRange() {
-        final String rangeString = getRange();
-        Validate.notNull(rangeString, "Range HTTP must not be null");
-        String[] rangeValuesStrings = StringUtils.split(rangeString, "bytes=");
-        Validate.isTrue(rangeValuesStrings.length == 1,
-                "Range header value doesn't begin with string: bytes=");
-
-        final String byteRange = rangeValuesStrings[0];
-
-        Validate.isTrue(StringUtils.split(byteRange, ",").length == 1,
-                "Multi-range requests are not supported");
-
-        String[] rangeParts = StringUtils.split(byteRange, "-");
-        Validate.isTrue(StringUtils.countMatches(byteRange, "-") < 2,
-                "Cannot end or start with a negative number");
-
-        Long startPos = null;
-        Long endPos = null;
-
-        if (StringUtils.startsWith(byteRange, "-")) {
-            endPos = Long.parseLong(byteRange);
-        } else if (StringUtils.endsWith(byteRange, "-")) {
-            startPos = Long.parseLong(byteRange.split("-")[0]);
-        } else if (rangeParts.length == 2) {
-            startPos = Long.parseUnsignedLong(rangeParts[0]);
-            endPos = Long.parseUnsignedLong(rangeParts[1]);
-        } else {
-            throw new IllegalArgumentException("range must exist with - separator");
-        }
-
-        return new Long[] {startPos, endPos};
+        return MantaUtils.parseSingleRange(getRange());
```

Comment:
This is now calling a deprecated method. Is that the intended behavior?

Reply:
The method was extracted into `MantaUtils` early on in the project but abandoned in favor of the `HttpRange` implementation. I'm putting the method body back and updating the JavaDoc to explain the "soft deprecation."